simplex/simplex_problem.py

```diff
@@ -0,0 +1,28 @@
+"""
+SimplexProblem and relevant functions.
+"""
+
+from dataclasses import dataclass
+from typing import Sequence
+
+import numpy as np
+
+from shared.constraints import SimplexConstraint
+from shared.minimization_problem import MinimizationProblem
+
+
+@dataclass
+class SimplexCallable:
```

Comment:
naming: SimplexCallable sounds like the general function, not a constraint
also, same information relevant for quadratic programming

maybe LinearConstraintCallable or so?